sonar-plugin-api/src/main/java/org/sonar/api/batch/rule/Rule.java

```diff
@@ -51,17 +50,9 @@
   RuleStatus status();
 
   /**
-   * Sub characteristic key.
-   *
-   * @since 4.3
-   */
-  @CheckForNull
-  String debtSubCharacteristic();
-
-  /**
-   * Remediation function : can by Linear (with a coefficient), Linear with offset (with a coefficient and an offset) or Constant per issue (with an offset)
-   *
    * @since 4.3
+   * @deprecated remediation function is not available in batch analyzer since version 5.2. Data computation is moved to server's Compute Engine.
+   * Calling this method throws an exception.
    */
   @CheckForNull
   DebtRemediationFunction debtRemediationFunction();
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @Deprecated annotation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)